dts/bindings/spi/spi.yaml

```diff
@@ -0,0 +1,31 @@
+---
+title: SPI Base Structure
+id: spi
+version: 0.1
+
+description: >
+    This binding gives the base structures for all SPI devices
+
+child:
+    bus: i2c
+
+properties:
+    "#address-cells":
+      type: int
+      category: required
+      description: should be 1.
+    "#size-cells":
+      type: int
+      category: required
+      description: should be 0.
+    clocks:
+      type: array
+      category: required
+      description: Clock gate information
```

Comment:
New to DT here, but I'm not sure from this description what should go into the clocks array.

Reply:
Also I don't see any clocks properties in any of the real device nodes defined later, even though this binding is `category: required`. What's wrong here?

Reply:
> New to DT here, but I'm not sure from this description what should go into the clocks array.

We should add a property in the SPI node to enable the SPI peripheral clock. Like it is done in u(s)art nodes (e.g. clocks = <&rcc STM32_CLOCK_BUS_APB2 0x00000020>). I used the same description which is used in uart and i2c yaml files. I plan to create PRs to add the clock property in SPI and I2C nodes for STM32 SOCs.

> Also I don't see any clocks properties in any of the real device nodes defined later, even though this binding is category: required. What's wrong here?

The python scripts which extracts the defines from the dts files does not check and force the required properties. Maybe, @erwango could share more info on this.

Reply:
Ack. https://github.com/zephyrproject-rtos/zephyr/issues/5460 was raised to track this issue.

Reply:
Yup, aware of the issue with not enforcing the check; my question is if it is really needed in this patch series since the clocks properties are not present in the SPI nodes. (And sorry if this is well understood from DT context).

From your example `<&rcc STM32_CLOCK_BUS_APB2 0x00000020>`, the contents are SoC specific (in this case, a phandle to the RCC node, parent clock bus ID, and bit mask).

Do we really need this property defined in the generic spi.yaml then, if the contents are always going to be opaque?

Reply:
> Yup, aware of the issue with not enforcing the check; my question is if it is really needed in this patch series since the clocks properties are not present in the SPI nodes. (And sorry if this is well understood from DT context).

I removed the clocks property because it isn't needed in this patch series.

> Do we really need this property defined in the generic spi.yaml then, if the contents are always going to be opaque?

I don't know how the SPI peripheral is implemented in each vendor's SOC. We may need to define clocks property in a vendor,spi.yaml file.

Reply:
Makes sense to me